corehq/apps/app_manager/static/app_manager/js/releases.js

```diff
@@ -253,12 +253,14 @@ hqDefine('app_manager/js/releases.js', function () {
             var lastApp = self.savedApps()[0];
             self.lastAppVersion(lastApp ? lastApp.version() : -1);
         });
-        self.url = function (name) {
-            var template = self.options.urls[name];
+        self.reverse = function() {
             for (var i = 1; i < arguments.length; i++) {
-                template = template.replace('___', ko.utils.unwrapObservable(arguments[i]));
+                arguments[i] = ko.utils.unwrapObservable(arguments[i]);
```

Comment:
Ha! That's kind of annoying that it does that unwrap thing rather than having the caller know whether their object needs to be unwrapped. I suppose you could push this out and just add in a `ko.utils.unwrapObservable` call wherever it's ambiguous (and there are many places where it's obvious, like when the arg is a string), and then you wouldn't need this self.reverse function at all—but probably you thought of that and thought against it as just not that much better for the amount of hassle, and I won't fault you for that :) Does just seem really pesky either way.